vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,236 @@
+            nodeFormation(request, spaceList, mapper, nodeArray, allNodeList, false);
+            for (Space space : spaceList) {
+                List<IModuleLinkDisplay> spaceToModulelinksList = allSpaceToModulelinksListMap.get(space.getId());
+                List<ISpaceLinkDisplay> spaceLinks = null;
+                spaceLinks = allSpaceToSpacelinksListMap.get(space.getId());
+                List<ISpaceLinkDisplay> spaceToSpaceLinksList = null;
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList = new ArrayList<>();
+                    spaceToSpaceLinksList.addAll(spaceLinks);
+                }
+                List<String> spaceAllLink = new ArrayList<>();
+                if (spaceToModulelinksList != null && !spaceToModulelinksList.isEmpty()) {
+
+                    spaceToModulelinksList.forEach(link -> {
+                        spaceAllLink.add(link.getLink().getModule().getId());
+                    });
+                }
+                if (spaceToSpaceLinksList != null && !spaceToSpaceLinksList.isEmpty()) {
+
+                    spaceToSpaceLinksList.forEach(link -> {
+                        spaceAllLink.add(link.getLink().getTarget().getId());
+                    });
+                }
+                spaceLinkMap.put(space.getId(), spaceAllLink);
+            }
+        }
+    }
+
+    /**
+     * To map each space to all linked space or each space to all linked module
+     * 
+     * @param <T>
+     * @param spaceToAllOtherlinksList
+     * @param spaceToAllOtherlinksMap
+     * @param isModule
+     */
+    private <T> void getSpaceToAllOtherlink(List<T> spaceToAllOtherlinksList,
+            Map<String, List<T>> spaceToAllOtherlinksMap, boolean isModule) {
+        spaceToAllOtherlinksList.forEach(link -> {
+            String spaceId = isModule == true ? ((IModuleLinkDisplay) link).getLink().getSpace().getId()
+                    : ((ISpaceLinkDisplay) link).getLink().getSourceSpace().getId();
+            if (!spaceToAllOtherlinksMap.containsKey(spaceId)) {
+                List<T> tempSpaceToOtherlinkList = new ArrayList<>();
+                spaceToAllOtherlinksMap.put(spaceId, tempSpaceToOtherlinkList);
+            }
+            List<T> tempSpaceToOtherlinkList = spaceToAllOtherlinksMap.get(spaceId);
+            tempSpaceToOtherlinkList.add(link);
+        });
+    }
+
+    /**
+     * creating json for space and module node in the spaceoverview graph
+     * 
+     * @param <T>
+     * @param request
+     * @param nodeList
+     * @param mapper
+     * @param nodeArray
+     * @param allNodeList
+     * @param isModule
+     */
+    private <T> void nodeFormation(HttpServletRequest request, List<T> nodeList, ObjectMapper mapper,
+            ArrayNode nodeArray, List<String> allNodeList, boolean isModule) {
+        nodeList.forEach(nodeval -> {
+            VSpaceElement nodeObj = isModule == true ? (Module) nodeval : (Space) nodeval;
```

Comment:
what's the point of this line? You are casting `nodeval` but you put it in a vspacelement variable.

Reply:
because VSpaceElement is the parent class of both Module and Space class. After ternary operator gets executed either it will Module Object or Space Object so it should be hold in a common variable.

Reply:
just pass VSpaceElement (which I believe should be IVspaceElement?) No casting needed.